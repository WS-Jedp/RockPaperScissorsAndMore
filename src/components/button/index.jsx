import React from 'react'
import { ButtonSecondaryWrapper, ButtonPrimaryWrapper } from './styles'

export const ButtonSecondary = ({title = 'Rules', action = () => {}}) => (
  <ButtonSecondaryWrapper onClick={action}>
    {title}
  </ButtonSecondaryWrapper>
)

export const ButtonPrimary = ({title = 'Play Again', action = () => {}}) => (
  <ButtonPrimaryWrapper onClick={action}>
    {title}
  </ButtonPrimaryWrapper>
)
