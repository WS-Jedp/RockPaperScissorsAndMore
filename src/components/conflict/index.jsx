import React from 'react'
import { ConflictWrapper } from './styles'

import { PickedOption } from '../pickedOption'
import { PlayAgain } from '../playAgain'

export const Conflict = ({userOption = {}, homeOption = {}}) => {

  const isMobile = false
  
  return (
    <>
      <ConflictWrapper>
        <PickedOption title='You Picked' option={userOption.name} color={userOption.color} picked={true} />
        {
        isMobile && <PlayAgain title='You Win'action={() => {}} />
        }
        <PickedOption title='The House Picked' option={homeOption.name} color={homeOption.color} />
      </ConflictWrapper>
      {
        !isMobile && <PlayAgain title='You Win'action={() => {}} />
      }
    </>
  )
}
