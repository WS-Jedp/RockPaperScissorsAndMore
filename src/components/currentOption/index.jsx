import React from 'react'
import { IconLizard, IconPaper, IconScissors, IconRock, IconSpock } from '../icons'

export const CurrentOption = ({option = 'rock'}) => {

  const optionName = option.toLowerCase(); 

    switch (optionName) {
      case "rock":
        return <IconRock />
        break;
      case "paper":
        return <IconPaper />
        break;
      case "scissors":
    return <IconScissors />
        break;
      case "lizard":
        return <IconLizard />
        break;
      case "spock":
        return <IconSpock />
        break;
    
      default:
        return <IconRock />
        break;
    }
  }