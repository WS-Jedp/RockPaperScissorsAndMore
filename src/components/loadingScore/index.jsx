import React from 'react'
import { LoadingWrapper } from './styles'

export const LoadingScore = () => (
  <LoadingWrapper />
)
