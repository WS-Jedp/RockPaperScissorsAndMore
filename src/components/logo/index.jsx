import React from 'react'
import logoSVG from '../../assets/images/logo-bonus.svg'

import { LogoWrapper } from './styles'

export const Logo = () => (
  <LogoWrapper aria-label="Logo Rock, Paper, Scissors And More">
    <img src={logoSVG} alt="Logo Rock, Paper, Scissors And More"/>
  </LogoWrapper>
)