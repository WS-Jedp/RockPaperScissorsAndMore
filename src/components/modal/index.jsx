import React from 'react'
import ReactDOM from 'react-dom'
import { ModalWrapper } from './styles'

export const Modal = ({children}) => ReactDOM.createPortal((
  <ModalWrapper>
    { children || 'Hello World' }
  </ModalWrapper>), 
  window.document.getElementById('modal-portal'))
