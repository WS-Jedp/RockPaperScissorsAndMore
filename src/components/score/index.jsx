import React from 'react'
import { ScoreWrapper } from './styles'

import { LoadingScore } from '../loadingScore'

export const Score = ({title = "score", score = 0, showScore}) => {

  return (
    <ScoreWrapper>
      <h2>{ title }</h2>
      <b>
        { showScore ? score : <LoadingScore /> }
      </b>
    </ScoreWrapper>
  )
}