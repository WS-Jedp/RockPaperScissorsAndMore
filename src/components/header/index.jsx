import React from 'react'
import { WrapperHeader } from './styles'

import { Score } from '../score/index'
import { Logo } from '../logo'

export const Header = () => (
  <WrapperHeader>
    <Logo />
    <Score />
  </WrapperHeader>
) 