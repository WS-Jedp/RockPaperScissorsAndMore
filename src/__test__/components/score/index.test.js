import React from 'react'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme'

import { Score } from '../../../components/score'

describe('Testing visual features of the component "Score"', () => {

  const ScoreComponent = create(<Score />)

  test('The visual component should be the same', () => {
    expect(ScoreComponent.toJSON()).toMatchSnapshot()
  }) 

  test('Should the title be a string', () => {
    expect(typeof mount(<Score />).find('h2').text()).toBe('string')
  })

  test('Should the score be a number', () => {
    expect(typeof Number(mount(<Score />).find('b').text())).toBe('number')
  })

  
})