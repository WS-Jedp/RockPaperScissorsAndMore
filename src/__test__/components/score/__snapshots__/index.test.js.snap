// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing visual features of the component "Score" The visual component should be the same 1`] = `
<div
  className="sc-AxjAm kusgnM"
>
  <h2>
    score
  </h2>
  <b>
    <div
      className="sc-AxirZ fztZbl"
    />
  </b>
</div>
`;
